src/SavedSearch.php

```diff
@@ -1174,53 +1191,37 @@ public static function dropdownDoCount(array $options = [])
     /**
      * Set do_count from massive actions
      *
-     * @param array   $ids      Items IDs
+     * @param integer $id     Item ID
      * @param integer $do_count One of self::COUNT_*
      *
      * @return boolean
      */
-    public function setDoCount(array $ids, $do_count)
+    public function setDoCount(int $id, $do_count)
     {
-        /** @var \DBmysql $DB */
-        global $DB;
-
-        $result = $DB->update(
-            $this->getTable(),
-            [
-                'do_count' => $do_count
-            ],
-            [
-                'id' => $ids
-            ]
-        );
+        $result = $this->update([
+            'id'       => $id,
+            'do_count' => $do_count
+        ]);
         return $result;
     }
 
 
     /**
      * Set entity and recursivity from massive actions
      *
-     * @param array   $ids   Items IDs
+     * @param integer   $id   Item ID
      * @param integer $eid   Entityy ID
      * @param boolean $recur Recursivity
      *
      * @return boolean
      */
-    public function setEntityRecur(array $ids, $eid, $recur)
+    public function setEntityRecur(int $id, $eid, $recur)
```

Comment:
You should not change the method signature, it can result in unexpected BC break in plugins.

IMHO, you can probably call the `update()` method directly from the massive action loop.